File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/speeds/aac/AAC4BHop.java
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.features.module.modules.movement.speeds.aac;

import net.ccbluex.liquidbounce.event.MoveEvent;
import net.ccbluex.liquidbounce.features.module.modules.movement.speeds.SpeedMode;
import net.ccbluex.liquidbounce.utils.MovementUtils;

public class AAC4BHop extends SpeedMode {
    private boolean legitHop;

    public AAC4BHop() {
        super("AAC4BHop");
    }

    @Override
    public void onMotion() {

    }

    @Override
    public void onUpdate() {

    }

    @Override
    public void onMove(MoveEvent event) {

    }

    @Override
    public void onEnable() {
        legitHop = true;
    }

    @Override
    public void onDisable() {
        mc.thePlayer.speedInAir = 0.02F;
    }

    @Override
    public void onTick() {
        if(MovementUtils.isMoving()) {
            if(legitHop) {
                if(mc.thePlayer.onGround) {
                    mc.thePlayer.jump();
                    mc.thePlayer.onGround = false;
                    legitHop = false;
                }
                return;
            }

            if(mc.thePlayer.onGround) {
                mc.thePlayer.onGround = false;
                MovementUtils.strafe(0.375F);
                mc.thePlayer.jump();
                mc.thePlayer.motionY = 0.41;
            }else
                mc.thePlayer.speedInAir = 0.0211F;
        }else{
            mc.thePlayer.motionX = 0D;
            mc.thePlayer.motionZ = 0D;
            legitHop = true;
        }
    }
}

File: shared/main/java/net/ccbluex/liquidbounce/features/module/modules/movement/speeds/aac/AAC4BHop.kt
/*
 * LiquidBounce Hacked Client
 * A free open source mixin-based injection hacked client for Minecraft using Minecraft Forge.
 * https://github.com/CCBlueX/LiquidBounce/
 */
package net.ccbluex.liquidbounce.features.module.modules.movement.speeds.aac

import net.ccbluex.liquidbounce.event.MoveEvent
import net.ccbluex.liquidbounce.features.module.modules.movement.speeds.SpeedMode
import net.ccbluex.liquidbounce.utils.MovementUtils

class AAC4BHop : SpeedMode("AAC4BHop") {
    private var legitHop = false

    override fun onDisable() {
        mc.thePlayer!!.speedInAir = 0.02f
    }

    override fun onTick() {
        val thePlayer = mc.thePlayer ?: return

        if (MovementUtils.isMoving) {
            if (legitHop) {
                if (thePlayer.onGround) {
                    thePlayer.jump()
                    thePlayer.onGround = false
                    legitHop = false
                }
                return
            }
            if (thePlayer.onGround) {
                thePlayer.onGround = false
                MovementUtils.strafe(0.375f)
                thePlayer.jump()
                thePlayer.motionY = 0.41
            } else thePlayer.speedInAir = 0.0211f
        } else {
            thePlayer.motionX = 0.0
            thePlayer.motionZ = 0.0
            legitHop = true
        }
    }

    override fun onMotion() {}
    override fun onUpdate() {}
    override fun onMove(event: MoveEvent) {}
    override fun onEnable() {
        legitHop = true
    }
}